### PR TITLE
examples/gkurve: initial commit for resizable_label

### PR DESCRIPTION
**Pull request by PiergiorgioZagaria**
- [X] By selecting this checkbox, I agree to license my contributions to this project under the license(s) described in the LICENSE file, and I have the right to do so or have received permission to do so by an employer or client I am producing work for whom has this right.

About the tessellator code, the tessellator.zig and data_structures come from https://github.com/fubark/cosmic, the tracy code from the zig compiler

### CODE REVIEW

File: examples/gkurve/LICENSE.tesselator [hunk 1]
```diff
@@ -0,0 +1 @@
+TODO: add license, the tesselator implementation comes from https://github.com/fubark/cosmic/blob/master/graphics/src/tessellator.zig
```
Comment (slimsag) [inline examples/gkurve/LICENSE.tesselator:1]: @fubark We're using your Cosmic tesselator implementation, and plan to continue doing so in the future. For now @PiergiorgioZagaria has extracted it from Cosmic so we can use it here, I'm curious how you'd feel about having this live in a separate repository/library proper?

(if that's just more work for you, no worries, just figured I'd ask!)

**Comment by slimsag** [comment]
This is super cool @PiergiorgioZagaria - I'm planning to review later in the week (probably in 2-3 days)

**Comment by PiergiorgioZagaria** [comment]
Is it ok if I make small commits to the PR for removing some bugs I found?

**Comment by slimsag** [comment]
Oh absolutely, push whatever improvements you like. I'll check them all out in a few days time.

**Review by slimsag — APPROVED** [review]
LGTM, there's definitely code cleanup I'd like to do here but it's fine - the solution looks great overall and I'm excited to have this.

Next up for me is (1) sorting out the ECS and (2) moving this logic into an ECS-backed solution

**Pull request merged**